### PR TITLE
Removed a number of unused fields, fixed analyzer warning for join

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/src/iterables/infinite_iterable.dart [hunk 1]
```diff
@@ -36,7 +36,7 @@ abstract class InfiniteIterable<T> extends IterableBase<T> {
 
   void forEach(void f(T element)) => throw new UnsupportedError('forEach');
 
-  String join([String separator]) => throw new UnsupportedError('join');
+  String join([String separator = ""]) => throw new UnsupportedError('join');
```
Comment (yjbanov) [inline lib/src/iterables/infinite_iterable.dart]: Use single-quotes for consistency

**Comment by yjbanov** [comment]
LGTM

**Comment by kevmoo** [comment]
Please merge - or give me commit access. :smile:

**Comment by yjbanov** [comment]
You should have access now. Try it by hitting the "merge" button on this PR :)

**Pull request merged**